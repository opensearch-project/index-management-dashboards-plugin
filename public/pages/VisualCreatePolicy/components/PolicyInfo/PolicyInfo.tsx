/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { ChangeEvent } from "react";
import {
  EuiSpacer,
  EuiCompressedFormRow,
  EuiCompressedFieldText,
  EuiTextArea,
  EuiPanel,
  EuiFlexGroup,
  EuiText,
  EuiHorizontalRule,
  EuiTitle,
} from "@elastic/eui";
import "brace/theme/github";
import "brace/mode/json";
import EuiFormCustomLabel from "../EuiFormCustomLabel";

interface PolicyInfoProps {
  isEdit: boolean;
  policyId: string;
  policyIdError: string;
  description: string;
  onChangePolicyId: (value: ChangeEvent<HTMLInputElement>) => void;
  onChangeDescription: (value: ChangeEvent<HTMLTextAreaElement>) => void;
}

const PolicyInfo = ({ isEdit, policyId, policyIdError, description, onChangePolicyId, onChangeDescription }: PolicyInfoProps) => (
  <EuiPanel>
    <EuiFlexGroup gutterSize="xs" alignItems="center">
      <EuiTitle size="s">
        <h2>{`Policy info`}</h2>
      </EuiTitle>
    </EuiFlexGroup>
    <EuiHorizontalRule margin={"xs"} />
    <EuiFormCustomLabel title="Policy ID" helpText="Specify a unique and descriptive ID that is easy to recognize and remember." />

    <EuiCompressedFormRow isInvalid={!!policyIdError} error={policyIdError}>
      <EuiCompressedFieldText
        disabled={isEdit}
        isInvalid={!!policyIdError}
        placeholder="hot_cold_workflow"
        readOnly={false}
        value={policyId}
        onChange={onChangePolicyId}
        data-test-subj="create-policy-policy-id"
      />
    </EuiCompressedFormRow>

    <EuiSpacer size="m" />

    <EuiFormCustomLabel title="Description" helpText="Describe the policy." />

    <EuiCompressedFormRow isInvalid={false} error={null}>
      <EuiTextArea
        style={{ minHeight: "150px" }}
        compressed={true}
        value={description}
        onChange={onChangeDescription}
        data-test-subj="create-policy-description"
      />
    </EuiCompressedFormRow>
  </EuiPanel>
);

export default PolicyInfo;
