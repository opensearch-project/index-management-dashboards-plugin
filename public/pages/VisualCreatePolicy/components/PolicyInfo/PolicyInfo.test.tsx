/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import { fireEvent } from "@testing-library/dom";
import userEvent from "@testing-library/user-event/dist";
import PolicyInfo from "./PolicyInfo";

describe("<PolicyInfo /> spec", () => {
  it("renders the component", () => {
    const { container } = render(
      <PolicyInfo
        isEdit={false}
        policyId="some_id"
        policyIdError=""
        description="some description"
        onChangePolicyId={() => {}}
        onChangeDescription={() => {}}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("calls on change policy id when typing in input", () => {
    const onChangePolicyId = jest.fn();
    const { getByTestId } = render(
      <PolicyInfo
        isEdit={false}
        policyId="some_id"
        policyIdError=""
        description="some description"
        onChangePolicyId={onChangePolicyId}
        onChangeDescription={() => {}}
      />
    );

    fireEvent.focus(getByTestId("create-policy-policy-id"));
    userEvent.type(getByTestId("create-policy-policy-id"), `some_policy_id`);
    expect(onChangePolicyId).toHaveBeenCalled();
  });

  it("calls on change description when typing in input", () => {
    const onChangeDescription = jest.fn();
    const { getByTestId } = render(
      <PolicyInfo
        isEdit={false}
        policyId="some_id"
        policyIdError=""
        description="some description"
        onChangePolicyId={() => {}}
        onChangeDescription={onChangeDescription}
      />
    );

    fireEvent.focus(getByTestId("create-policy-description"));
    userEvent.type(getByTestId("create-policy-description"), `some description`);
    expect(onChangeDescription).toHaveBeenCalled();
  });
});
