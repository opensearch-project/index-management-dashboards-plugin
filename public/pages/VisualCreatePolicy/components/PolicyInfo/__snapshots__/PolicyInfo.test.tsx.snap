// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PolicyInfo /> spec renders the component 1`] = `
<div
  class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
  style="padding-left: 0px; padding-right: 0px;"
>
  <div
    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsFlexStart euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
    style="padding: 0px 10px;"
  >
    <div
      class="euiFlexItem"
    >
      <h3
        class="euiTitle euiTitle--small"
      >
        Policy info
        <span
          class="panel-header-count"
        >
           
           
        </span>
      </h3>
    </div>
  </div>
  <hr
    class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
  />
  <div
    style="padding: initial;"
  >
    <div
      style="padding: 10px 0px 0px 10px;"
    >
      <div
        class="euiText euiText--medium"
        style="margin-bottom: 5px;"
      >
        <h5
          class="euiFormLabel "
          style="margin-bottom: 2px;"
        >
          Policy ID
        </h5>
        <p>
           
          <span
            style="font-weight: 200; font-size: 12px;"
          >
            Specify a unique and descriptive ID that is easy to recognize and remember.
             
          </span>
        </p>
      </div>
      <div
        class="euiFormRow"
        id="some_html_id-row"
      >
        <div
          class="euiFormRow__fieldWrapper"
        >
          <div
            class="euiFormControlLayout"
          >
            <div
              class="euiFormControlLayout__childrenWrapper"
            >
              <input
                class="euiFieldText"
                data-test-subj="create-policy-policy-id"
                id="some_html_id"
                placeholder="hot_cold_workflow"
                type="text"
                value="some_id"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--m"
      />
      <div
        class="euiText euiText--medium"
        style="margin-bottom: 5px;"
      >
        <h5
          class="euiFormLabel "
          style="margin-bottom: 2px;"
        >
          Description
        </h5>
        <p>
           
          <span
            style="font-weight: 200; font-size: 12px;"
          >
            Describe the policy.
             
          </span>
        </p>
      </div>
      <div
        class="euiFormRow"
        id="some_html_id-row"
      >
        <div
          class="euiFormRow__fieldWrapper"
        >
          <textarea
            class="euiTextArea euiTextArea--resizeVertical euiTextArea--compressed"
            data-test-subj="create-policy-description"
            id="some_html_id"
            rows="3"
            style="min-height: 150px;"
          >
            some description
          </textarea>
        </div>
      </div>
    </div>
  </div>
</div>
`;
